### PR TITLE
first attempts at cleaning up fluxController

### PR DESCRIPTION
**Pull request by pineapplespatula**
this PR adds the ability to flag actions as private
it also adds a module tag to each action in flux controller, and filters out functions that are not actions from being called.

This may cause widespread side effects depending on how consitantly we implemented and used actions previously so I'm marking it as review only for now

### CODE REVIEW

File: src/js/fluxcontroller.js [hunk 1]
```diff
@@ -77,6 +77,11 @@ define(function (require, exports, module) {
     var MAX_RETRY_WINDOW = 6400;
 
     /**
+    *@type {Object}
```
Comment (iwehrman) [inline src/js/fluxcontroller.js]: Broken comment formatting, plus this variable should be part of the object state. And private.

File: src/js/fluxcontroller.js [hunk 2]
```diff
@@ -446,6 +472,15 @@ define(function (require, exports, module) {
                         throw new Error(message);
                     }
     
+                    if (nextAction.private) {
+                        if (nextAction.module !== action.module) {
+                            var privateMessage = "Invalid transfer from " + actionName + " to " + nextActionName +
+                                    ". " + nextActionName + " is declaed private and is in a different module from" +
```
Comment (iwehrman) [inline src/js/fluxcontroller.js]: Typo.

File: src/js/fluxcontroller.js [hunk 3]
```diff
@@ -588,6 +623,9 @@ define(function (require, exports, module) {
             actionTitle = "sub-action " + actionName + " of action " + parentActionName;
         } else {
             actionTitle = "action " + actionName;
+            if (action.private) {
+                throw new Error("Action " + actionName + "is private and cannot be exectued");
```
Comment (iwehrman) [inline src/js/fluxcontroller.js]: Typo.

File: src/js/fluxcontroller.js [hunk 4]
```diff
@@ -202,6 +208,26 @@ define(function (require, exports, module) {
...
+     * @private
+     */
+    FluxController.prototype._initAllActions = function () {
+        Object.keys(actionIndex).forEach(function (actionModuleName) {
+            var actionModule = actionIndex[actionModuleName],
+                moduleActions = {};
+            Object.keys(actionModule)
+                .forEach(function (actionName) {
+                    if (actionModule[actionName].action) {
+                        actionModule[actionName].module = actionModuleName;
```
Comment (iwehrman) [inline src/js/fluxcontroller.js]: I don't think we should modify the `.action` object. Instead, we should just keep this state in the `FluxController` instances, the same way we keep the locks and the name/function mappings.

File: src/js/actions/documents.js [hunk 5]
```diff
@@ -304,7 +304,8 @@ define(function (require, exports) {
     };
     initInactiveDocuments.action = {
         reads: [locks.PS_DOC],
-        writes: [locks.JS_DOC]
+        writes: [locks.JS_DOC],
+        private: true
```
Comment (iwehrman) [inline src/js/actions/documents.js]: :white_check_mark:

**Comment by iwehrman** [comment]
Nice start!

**Comment by pineapplespatula** [comment]
@iwehrman  :repeat_one:

File: src/js/fluxcontroller.js [hunk 6]
```diff
@@ -202,6 +219,32 @@ define(function (require, exports, module) {
     });
 
     /**
+     * Initialize all action functions 
+     *
+     * @private
+     */
+    FluxController.prototype._initAllActions = function () {
+        this._allActions = {};
```
Comment (iwehrman) [inline src/js/fluxcontroller.js]: This should be a `Map`.

File: src/js/fluxcontroller.js [hunk 7]
```diff
@@ -202,6 +219,32 @@ define(function (require, exports, module) {
...
+        this._actionModules = new Map();
```
Comment (iwehrman) [inline src/js/fluxcontroller.js]: I think this should be called `_actionModuleNames`? It maps actions to action module names, right? The JSDoc above should be clarified too.

**Comment by iwehrman** [comment]
Is this supposed to be review-only, @pineapplespatula?

**Comment by pineapplespatula** [comment]
since its such a broad change i would want a QE pass before it gets merged. if there is a better flag for that let me know

**Comment by iwehrman** [comment]
Review-only is great, but you should assign it to @chadrolfs or @ktaki. Also, you might as well mark everything else as private now that should be private?

**Comment by pineapplespatula** [comment]
@ktaki  can you give this branch a once over to see if i broke anything unexpected

**Comment by ktaki** [comment]
@pineapplespatula Do you have any suggestion how your changes be tested? Any particular areas of concerns?

**Comment by iwehrman** [comment]
@ktaki @pineapplespatula: I don't think this would cause any subtle bugs. If it seems to be working then it's probably working. I suggest you annotate the other private actions and we get on with it.

**Comment by ktaki** [comment]
@pineapplespatula: You told me to hold testing while you were making some more changes. Let me know if you've done with it.

**Comment by pineapplespatula** [comment]
Closing this PR

**Comment by iwehrman** [comment]
Out of curiosity, why did you decide to close this instead of finishing it up?

**Comment by pineapplespatula** [comment]
There aren't really any truly "private" actions, but there are a number of actions that should be "protected" however the amount of cleanup around the actions to stop exporting these functions which are not actions seemed a larger scope than i wanted to bite off for something that doesn't directly add to our goals

**Pull request closed without merging**